File: src/micapolos/tata8/Animation.java
package micapolos.tata8;

public interface Animation {
  void update(float seconds);

  Animation EMPTY = _ -> {};

  static Animation move(Position position, Speed speed) {
    return seconds -> {
      position.x += speed.x * seconds;
      position.y += speed.y * seconds;
    };
  }
}

File: src/micapolos/zexy/indexed/Animation.kt
package micapolos.zexy.indexed

sealed class Animation: Value {
  object Empty : Animation()
  class Once(val action: Value) : Animation()
  class EveryStep(val action: Value): Animation()
  class Pause(val seconds: Value): Animation()
  class Parallel(val animations: List<Value>): Animation()
  class Race(val animations: List<Value>): Animation()
  class Sequence(val animations: List<Value>): Animation()
  class RepeatWhile(val animation: Value, val condition: Value): Animation()
  class On(val trigger: Value, val animation: Value): Animation()
}